examples/stable_diffusion/notebook/sd_multilora.ipynb

```diff
@@ -0,0 +1,431 @@
+    "    [prompt] * batch_size,\n",
+    "    num_inference_steps=20,\n",
+    "    height=512,\n",
+    "    width=512,\n",
+    ")\n",
+    "\n",
+    "for image_index in range(batch_size):\n",
+    "    output_path = f\"result_wolf_{image_index}.png\"\n",
+    "    result.images[image_index].save(output_path)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Then you can see the image, like this:   \n",
+    "![wolf plushie](image/result_wolf.png)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Finally, extract the adapters from the ONNX model."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Extract adapters from pen sketch unet model\n",
+    "!olive generate-adapter -m onnx_model/unet_pen_sketch -o adapters/pen_sketch\n",
+    "# Extract adapters from wolf plushie unet model\n",
+    "!olive generate-adapter -m onnx_model/unet_wolf_plushie -o adapters/wolf_plushie"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Deploy Model with Multiple Adapters\n",
+    "\n",
+    "We can now deploy the same model with multiple adapters for different tasks by loading the adapter weights independently of the model and providing the relevant weights as input at inference time."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import shutil\n",
+    "\n",
+    "file_list = [\"model.onnx\", \"model.onnx.data\"]\n",
+    "old_model_path = Path(\"adapters/pen_sketch/model\")\n",
+    "base_model_path = Path(\"model\")\n",
+    "\n",
+    "base_model_path.parent.mkdir(parents=True, exist_ok=True)\n",
+    "\n",
+    "# copy base onnx model to new folder\n",
+    "for file in file_list:\n",
+    "    shutil.copy2(old_model_path / file, base_model_path / file)\n",
+    "\n",
+    "base_model_path = base_model_path / \"model.onnx\"\n",
+    "\n",
+    "# load unet session from base model\n",
+    "unet_session = OnnxRuntimeModel.load_model(base_model_path, provider=provider)\n",
+    "\n",
+    "adapters = {\n",
+    "    \"pen_sketch\":  \"adapters/pen_sketch/model/adapter_weights.onnx_adapter\",\n",
+    "    \"wolf_plushie\": \"adapters/wolf_plushie/model/adapter_weights.onnx_adapter\"\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Here, we implement a `UNetSessionWrapper` to inject adapter weights as inputs to the UNet model."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import sys\n",
+    "\n",
+    "utils_path = Path().resolve().parent.parent.parent / \"olive\" / \"common\"\n",
+    "sys.path.append(str(utils_path))\n",
+    "\n",
+    "from utils import load_weights\n",
+    "\n",
+    "class UNetSessionWrapper:\n",
```

Comment:
Since the main benefit of multi-lots is that we can use the same session/pipeline for different adapters, could you make this something like 

```

class UNetSessionWrapper:
    def __init__(self, unet_session):
        self.unet_session = unet_session
        self.adapters = {}
        self.active_adapter = None

    def load_adapter(self, adapter_name, adapter_path):
        self.adapters[adapter_name] = load_weights(adapter_path)

    def set_adapter(self, adapter_name):
        assert adapter_name in self.adapters, f"Adapter {adapter_name} not found"
        self.active_adapter = adapter_name

    def unset_adapter(self):
        self.active_adapter = None

    def run(self, output_names, input_feed):
        # running when adapter is not set is equivalent to running the base model
        inputs = {**input_feed, **self.adapters.get(self.active_adapter, {})}
        return self.unet_session.run(output_names, inputs)

    def __getattr__(self, name):
        return getattr(self.unet_session, name)
```

Reply:
Thanks!